utils/common/src/main/java/brooklyn/util/text/Strings.java

```diff
@@ -136,26 +150,45 @@ public static String removeFromEnd(String string, String ...suffixes) {
         return string;
     }
 
-    /** as removeFromEnd, but repeats until all such suffixes are gone */
-    public static String removeAllFromEnd(String string, String ...suffixes) {
+    /**
+     * As removeFromEnd, but repeats until all such suffixes are gone
+     */
+    public static String removeAllFromEnd(String string, String... suffixes) {
+        if (isEmpty(string)) return string;
+        int index = string.length();
         boolean anotherLoopNeeded = true;
         while (anotherLoopNeeded) {
             if (isEmpty(string)) return string;
```

Comment:
The first check is really for `string != null`, so that `.length()` doesn't throw an exception.